client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala

```diff
@@ -23,23 +21,36 @@ object FollowRedirect {
                 fragment = uri.fragment orElse req.uri.fragment
               )
 
-              prepareLoop(
-                req.copy(method = method, uri = nextUri, body = EmptyBody),
-                redirects + 1
-              )
+              // We're following a redirect, so we need to dispose of the last response
+              dispose.flatMap { _ =>
+                prepareLoop(
+                  req.copy(method = method, uri = nextUri, body = EmptyBody),
+                  redirects + 1
+                )
+              }
 
             case _ => Task.now(dr)
           }
         }
 
         resp.status.code match {
-          // We cannot be sure what will happen to the request body so we don't attempt to deal with it
-          case 301 | 302 | 307 | 308 if req.body.isHalt =>
```

Comment:
That sounds right, in which case I think it would be better to `uncons` to verify that the body was strict.  Otherwise, we can only do uninteresting re-POSTs.

Automatically re-POSTing is starting to feel like it's on shaky ground anyway.